pkg/crc/machine/start.go

```diff
@@ -390,19 +394,20 @@ func (client *client) Start(ctx context.Context, startConfig types.StartConfig)
 		return nil, errors.Wrap(err, "Error waiting for apiserver")
 	}
 
-	// Remove this check when we have official 4.8 bundle
-	if strings.HasPrefix(crcBundleMetadata.GetOpenshiftVersion(), "4.8.") {
-		if err := cluster.EnsureSSHKeyPresentInTheCluster(ocConfig, constants.GetPublicKeyPath()); err != nil {
-			return nil, errors.Wrap(err, "Failed to update ssh public key to machine config")
-		}
+	if err := cluster.EnsurePullSecretPresentInTheCluster(ocConfig, startConfig.PullSecret); err != nil {
+		return nil, errors.Wrap(err, "Failed to update cluster pull secret")
 	}
 
-	if err := ensureProxyIsConfiguredInOpenShift(ocConfig, sshRunner, proxyConfig, instanceIP); err != nil {
-		return nil, errors.Wrap(err, "Failed to update cluster proxy configuration")
+	if err := cluster.EnsureSSHKeyPresentInTheCluster(ocConfig, constants.GetPublicKeyPath()); err != nil {
+		return nil, errors.Wrap(err, "Failed to update ssh public key to machine config")
 	}
 
-	if err := cluster.EnsurePullSecretPresentInTheCluster(ocConfig, startConfig.PullSecret); err != nil {
-		return nil, errors.Wrap(err, "Failed to update cluster pull secret")
+	if err := cluster.WaitForPullSecretPresentOnInstanceDisk(sshRunner); err != nil {
```

Comment:
I am wondering if we could move that just before `cluster.WaitForClusterStable`.
So that we save a bit of time: every config changes are sent, we are just waiting for them to be applied.

Reply:
@guillaumerose I can try but doesn't the proxy test are failing because of that, I had to put it above proxy configuration.

Reply:
ok np, let's start with this.